File: 프로그래머스/2/42839. 소수 찾기/소수 찾기.py
from itertools import permutations

def isPrime(n):
    if n == 0 or n == 1: return False
    for i in range(2,n):
        if n%i == 0: return False
    return True

def solution(numbers):
    answer = 0
    pers = set()
    for l in range(1,len(numbers)+1):
        temp = list(permutations(list(numbers),l))
        for p in temp: pers.add(int(''.join(p)))
    for p in pers:
        if isPrime(p): answer += 1
    return answer

File: 프로그래머스/1/12921. 소수 찾기/소수 찾기.java
class Solution {
    public int solution(int n) {
        int answer = 0;
        boolean[] ch = new boolean[n+1];
        for(int i=2; i<=n; i++) {
            if(!ch[i]) {
                answer++;
                for(int j=i; j<=n; j=j+i) ch[j] = true;
            }
        }
        return answer;
    }
}